module/zfs/zil.c

```diff
@@ -337,6 +354,85 @@ zil_read_log_data(zilog_t *zilog, const lr_write_t *lr, void *wbuf)
 	return (error);
 }
 
+void
+zil_set_kstat_ptr(zilog_t *zilog, zil_sums_t *zsum)
+{
+	zilog->zl_sums = zsum;
+}
+
+void
+zil_copy_kstat_values(zil_stats_t *zs)
+{
+	memcpy(zs, &zil_stats, sizeof (zil_stats));
```

Comment:
Here you are copying global statistics into dataset one.  It is OK until there appear some fields that are not set by zil_update_wmsum().

Reply:
@amotin - If I understand you correctly, We are using same ```zil_stats_t``` structure for both global and per dataset, and we are using ```zil_update_wmsum``` for both the cases. So any change in ```zil_stats_t``` will require changes in ```zil_update_wmsum()``` also.

Reply:
Yes. But I am talking about potential fields that may still use atomics or somehow else we might not inherit from global to dataset.